packages/cardano-services/src/util/TypeormProvider/TypeormProvider.ts

```diff
@@ -22,6 +22,19 @@ export abstract class TypeormProvider extends TypeormService implements Provider
   }
 
   async healthCheck(): Promise<HealthCheckResponse> {
+    try {
+      if (this.state === 'running') {
+        const dataSource = await firstValueFrom(this.dataSource$);
+
+        if (!dataSource) throw new Error('No DataSource');
```

Comment:
Is there any point of calling `onError` in this case? It should already be (re)connecting when it's already `null`. This could potentially mess up the re-connection logic. I suggest returning unhealthy here instead of throwing.

Reply:
a9ccfea1b0 should solve this